stmtools/_io.py

```diff
@@ -0,0 +1,137 @@
+          Defaults to None, then the following renaming will be performed: 
+          "{"pnt_lat": "lat", "pnt_lon": "lon"}"
+        output_chunksize (dict | None, optional): Chunksize of the output. Defaults to None, 
+          then the size of the first chunk in the DaskDataFrame will be used, up-rounding to 
+          the next 5000.
+        blocksize (int | str | None, optional): Blocksize to load the csv. 
+          Defaults to 200e6 (in bytes). See the documentation of 
+          [dask.dataframe.read_csv](https://docs.dask.org/en/stable/generated/dask.dataframe.read_csv.html)
+
+    Returns:
+        xr.Dataset: Output STM instance
+    """
+
+    # Load csv as Dask DataFrame
+    ddf = dd.read_csv(file, blocksize=blocksize)
+
+    # Take the first column and compute the chunk sizes
+    da_col0 = ddf[ddf.columns[0]].to_dask_array()
+    da_col0.compute_chunk_sizes()
+    chunks = da_col0.chunks
+
+    # Count time dimension by one spacetime column
+    if spacetime_pattern is None:
+        spacetime_pattern = {"^d_": "deformation", "^a_": "amplitude", "^h2ph_": "h2ph"}
+    time_shape = 0
+    if spacetime_pattern is not None:
+        key = list(spacetime_pattern.keys())[0]
+        for column in ddf.columns:
+            if re.match(re.compile(key), column):
+                time_shape += 1
+
+    # Initiate a template STM
+    coords = {
+        "space": range(da_col0.shape[0]),
+        "time": range(time_shape),
+    }
+    stmat = xr.Dataset(coords=coords)
+
+    # Read csv by columns
+    dict_temp_da = dict()  # Temporary dask array collector
+    for k in spacetime_pattern.keys():
+        dict_temp_da[k] = []
+
+    # Temporaly save time-series columns to lists in dict_temp_da
+    for column in ddf.columns:
+        if _is_space(space_pattern, column):
+            da_pnt = ddf[column].to_dask_array(lengths=chunks[0])
+            stmat = stmat.assign({column: (("space"), da_pnt)})
+        else:
+            for k in spacetime_pattern.keys():
+                if re.match(re.compile(f"{k}"), column):
+                    da_list = dict_temp_da[k]
+                    da_list.append(ddf[column].to_dask_array(lengths=chunks[0]))
+                    dict_temp_da[k] = da_list
+
+    # Stack dask arrays in dict_temp_da, assign to STM
+    for k, da_list in dict_temp_da.items():
+        da_ts = da.stack(da_list)  # Stack on time dimension
+        stmat = stmat.assign({spacetime_pattern[k]: (("time", "space"), da_ts)})
+
+    # Rearrage space and time order
+    stmat = stmat.transpose("space", "time")
+
+    # Uniform chunking
+    if output_chunksize is None:
+        output_chunksize = chunks[0][0]
+        if len(chunks[0]) > 1:
+            # If more than one chunk, use the first chunk size, round to next 5000
+            output_chunksize = _round_chunksize(output_chunksize)
+    stmat = stmat.chunk(output_chunksize)
+
+    # Set coordinates
```

Comment:
When creating STMs from SLC files you end up with a Dataset that has time and points dimension, while here you have time and space. Should one maybe use the same notation in both cases? 

Also, the amplitude and phase are treated as coordinates in the SLC case, while they are assigned as variables here, should one treat them in the same way on both sides?

Reply:
Hi @fnattino, we are working on renaming the dimension name to "space". See #35.

Reply:
Ah right, sorry, I got confused. Nice you are already considering renaming 'points' to 'space'.